emrichen/template.py

```diff
@@ -44,11 +44,14 @@ def parse(cls, data, format=None, filename=None):
         return cls(template=parse(data, format=format), filename=filename)
 
 
-def extract_defaults(template):
-    defaults = {}
+def extract_defaults(template, filename):
+    defaults = dict(__file__=filename)
+
     for doc in template:
         if isinstance(doc, Defaults):
             defaults.update(doc.data)
+        elif isinstance(doc, Include):
```

Comment:
Hnnf, shouldn't this recurse down any chain of includes?

Reply:
Doesn't it? It delegates to `Include.get_template`, which delegates to `Template.parse` and `Template.__init__`, which finally calls `extract_defaults` again.

Reply:
Added a test, works. f35a7c2